### PR TITLE
Reduce circleci parallelism

### PR DESCRIPTION
**Pull request by egh**
- Reduce parallelism on circleci
- Set `BUILD_EMBER` on circle
- Bump max wait time to 15 :cry: 
- reset cache version to try to fix flaky specs
- Change `test.rb` config to raise errors in the server when testing with capbara
- Many other feature spec fixes to reduce/eliminate various flaky spec failures

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -45,12 +46,12 @@ jobs:
 
     steps:
       - restore_cache:
-          key: source-v6
+          key: source-v7
```
Comment (slimeate) [inline .circleci/config.yml]: What's this incrementing about? Do we have to create a new cache key every time there's a config.yml change or something?

File: spec/features/front_matter_reviewer_report_spec.rb [hunk 2]
```diff
@@ -67,189 +67,192 @@ def create_reviewer_report_task
...
   end
 
-  scenario 'A reviewer can fill out their own Reviewer Report, submit it, and see a readonly view of their responses' do
-    create_reviewer_invitation(paper)
-    reviewer_report_task = create_reviewer_report_task
+  context 'reviewer is logged in' do
+    before do
+      login_as(reviewer, scope: :user)
+      visit "/"
+    end
```
Comment (slimeate) [inline spec/features/front_matter_reviewer_report_spec.rb]: This is the only meaningful change in this file right? Everything else is just indentation noise?

File: spec/features/initial_decision_spec.rb [hunk 3]
```diff
@@ -24,8 +24,8 @@
 include RichTextEditorHelpers
 
 feature 'Initial Decision', js: true, sidekiq: :inline! do
-  given(:admin) { FactoryGirl.create(:user, :site_admin) }
-  given(:paper) do
```
Comment (slimeate) [inline spec/features/initial_decision_spec.rb]: I had no idea

**Comment by benjaminkreen** [comment]
bless your heart @egh this looks like a ton of work

File: .circleci/config.yml (hunk 1 above)
Comment (egh) [reply .circleci/config.yml]: No, but this is the only way to clear the cache now. I'm not sure how often we need to clear the cache. I figured it wouldn't hurt.

File: spec/features/initial_decision_spec.rb (hunk 3 above)
Comment (egh) [reply spec/features/initial_decision_spec.rb]: Me neither. It's supposed to be equivalent to `let` but it seemed to fix an issue.

File: spec/support/tahi_helper_methods.rb [hunk 4]
```diff
@@ -79,6 +79,11 @@ def with_aws_cassette(name)
     end
   end
```
Comment (slimeate) [inline spec/support/tahi_helper_methods.rb]: Can we get rid of this helper now?

Comment (egh) [reply spec/support/tahi_helper_methods.rb]: It's still used. I only created the `insert_aws_cassette` helper to work around an issue in rspec-capybara. capybara does not work with `around` blocks so in https://github.com/Aperta-project/Aperta/pull/4091/files#diff-01359a56dac986586ece9a7c9bb23676R81 where I abstracted out the various combinations of preprint flag, etc I needed to use `before` and `after` to insert/eject VCR cassettes.

Comment (egh) [reply spec/support/tahi_helper_methods.rb]: https://github.com/teamcapybara/capybara/issues/1029

File: spec/features/front_matter_reviewer_report_spec.rb (hunk 2 above)
Comment (egh) [reply spec/features/front_matter_reviewer_report_spec.rb]: Yeah, sorry...

**Comment by egh** [comment]
@benjaminkreen Thanks, Ben. I'm on a mission

File: spec/support/tahi_helper_methods.rb (hunk 4 above)
Comment (slimeate) [reply spec/support/tahi_helper_methods.rb]: Ah, got it

File: spec/features/research_article_reviewer_report_spec.rb [hunk 5]
```diff
@@ -84,185 +82,191 @@ def create_reviewer_report_task
...
-    ctt = CardTaskType.find_by(task_class: "TahiStandardTasks::UploadManuscriptTask")
-    FactoryGirl.create(:card, :versioned, card_task_type: ctt)
-
-    create_reviewer_invitation(paper)
-    reviewer_report_task = create_reviewer_report_task
+  context 'reviewer is logged in' do
+    before do
+      login_as(reviewer, scope: :user)
+      visit "/"
+    end
```
Comment (slimeate) [inline spec/features/research_article_reviewer_report_spec.rb]: Also assuming this is the only real change here in this file.

Comment (egh) [reply spec/features/research_article_reviewer_report_spec.rb]: Correct.

Comment (egh) [reply spec/features/research_article_reviewer_report_spec.rb]: Well, there were a few things to make rubocop happy.

File: spec/support/pages/admin_dashboard_page.rb [hunk 6]
```diff
@@ -129,13 +129,16 @@ def username=(attr)
 
   def save
     click_on "Save"
-    wait_for_ajax
-    AdminDashboardPage.new(context: context)
+    AdminDashboardPage.new(context: context).tap do |page|
+      expect(page).not_to have_css('.overlay-container')
```
Comment (slimeate) [inline spec/support/pages/admin_dashboard_page.rb]: might be good to just add this to the constructor for AdminDashboardPage.

**Comment by slimeate** [comment]
Looks good to me! One open comment about moving that timing code into the page object constructor, but I don't think it's a big deal.

File: spec/support/pages/admin_dashboard_page.rb (hunk 6 above)
Comment (egh) [reply spec/support/pages/admin_dashboard_page.rb]: this method is actually on an edit modal overlay for the admin dashboard page. On calling the `save` or `cancel` methods it returns a new `AdminDashboardPage`. Adding this to the constructor would move this code away from the place where it is actually needed, when the user has taken an action to close the overlay.

**Comment by egh** [comment]
Gonna run this a bunch of times and see if it stays green

File: spec/features/research_article_reviewer_report_spec.rb [hunk 7]
```diff
@@ -261,8 +261,7 @@ def create_reviewer_report_task
       wait_for_editors
       set_rich_text editor: 'reviewer_report--competing_interests--detail', text: 'save this'
       find('.required-standalone textarea').set 'Edit notes'
-      wait_for_ajax
-      expect(Answer.first.value).to eq '<p>I have no competing interests</p>'
+      expect { Answer.first.value }.to become('<p>I have no competing interests</p>')
```
Comment (slimeate) [inline spec/features/research_article_reviewer_report_spec.rb]: `Answer.first` wont cache right?

Comment (egh) [reply spec/features/research_article_reviewer_report_spec.rb]: you know I was wondering that

Comment (egh) [reply spec/features/research_article_reviewer_report_spec.rb]: no
```
[5] pry(main)> User.first.object_id
=> 46963211519280
[6] pry(main)> User.first.object_id
=> 46963212048300
```

Comment (slimeate) [reply spec/features/research_article_reviewer_report_spec.rb]: does it cache the sql query though?

Comment (egh) [reply spec/features/research_article_reviewer_report_spec.rb]:
```
require "rails_helper"

RSpec.describe 'misc' do
  let!(:journal) { FactoryGirl.create(:journal) }

  it 'should not cache db queries' do
    expect do
      Journal.first.name
      Journal.first.name
      Journal.first.name
    end.to make_database_queries(count: 3)
  end
end
```
```
Finished in 0.04366 seconds (files took 9.92 seconds to load)
1 example, 0 failures
```

No.

Comment (egh) [reply spec/features/research_article_reviewer_report_spec.rb]: And double checking the log:
```
  Journal Load (0.8ms)  SELECT  "journals".* FROM "journals"  ORDER BY "journals"."id" ASC LIMIT 1
  Journal Load (0.5ms)  SELECT  "journals".* FROM "journals"  ORDER BY "journals"."id" ASC LIMIT 1
  Journal Load (0.4ms)  SELECT  "journals".* FROM "journals"  ORDER BY "journals"."id" ASC LIMIT 1
```

Comment (slimeate) [reply spec/features/research_article_reviewer_report_spec.rb]: Nice, thanks for checking. To this day, I'm still woefully uninformed about how AR SQL queries are cached. Thankfully it appears to work how I'd expect in rspec--that is, it doesn't cache.

Comment (egh) [reply spec/features/research_article_reviewer_report_spec.rb]: I believe this is because controller caching is disabled in test env

Comment (slimeate) [reply spec/features/research_article_reviewer_report_spec.rb]: right, and this is also outside of a controller.

**Comment by egh** [comment]
@slimeate I'm going to say this is "good enough". Our green/red ratio isn't great but the test failures now seem to be genuine timeouts.
selection_222

**Comment by slimeate** [comment]
I've been paying attention to the commits coming in since I last reviewed. Everything looks good.

"Good enough" is probably a good place to stop. This is certainly a huge improvement.

**Comment by egh** [comment]
:+1: Thanks, @slimeate !!

**Pull request merged**